### DISCUSSION TITLE
How to depoly or use the flask app with the machine learning model

### DISCUSSION THREAD

**Discussion by vivianc2** [CornellDataScience/MathSearch#40] [Q&A] [answered]
How to depoly or use the flask app with the machine learning model？

**Comment by emerisly** [comment, marked as answer]
I recommend [this tutorial](https://medium.com/techfront/step-by-step-visual-guide-on-deploying-a-flask-application-on-aws-ec2-8e3e8b82c4f7) from medium for a simple flask app.

## To deploy
Please do not run this command unless you know what you are doing
```sudo systemctl restart <NAME.service>; sudo systemctl enable <NAME.service>; sudo systemctl restart nginx```

## To debug with logs on localhost
cd into dir: ```cd ~/web```
run flask with gunicorn on localhost: ```/opt/conda/envs/pytorch/bin/gunicorn -b localhost:5000 app:app```
run url, example: ```curl localhost:5000/coord```
